pandas/core/series.py

```diff
@@ -2461,6 +2461,10 @@ def asof(self, where):
 
         If there is no good value, NaN is returned.
 
+        Note that this is really just a convenient shorthand for `Series.reindex`,
+        and is equivalent to `s.dropna().reindex(where, method='ffill')` for
+        an array of dates.
```

Comment:
Yes, but with the new implementation this equivalence holds for all index types I think.

The only difference is that `asof` tries to parse strings as dates, and `reindex` not.